syntax = "proto3";

package PythonMakeCNNModel;

service PythonMakeCNNModelService {
    rpc Start(StartRequest) returns (stream StartResponse);
}

message StartRequest {
    repeated string modes = 1;
    string save_model_path = 2;
    int32 epochs = 3;
    int32 batch_size = 4;
    bool enable_logging = 5;
}

message StartResponse {
    bool result = 1;
    string error_message = 2;
    string status = 3;
    string progress = 4;
}